internal/controller/finbackup_controller.go

```diff
@@ -666,13 +668,184 @@ func (r *FinBackupReconciler) createOrUpdateBackupJob(
+			RunAsUser:    ptr.To(int64(10000)),
+		}
+
+		job.Spec.Template.Spec.ServiceAccountName = "fin-deletion-job"
+
+		job.Spec.Template.Spec.RestartPolicy = corev1.RestartPolicyOnFailure
+
+		job.Spec.Template.Spec.Containers = []corev1.Container{
+			{
+				Name:    "deletion",
+				Command: []string{"/manager"},
+				Args:    []string{"deletion"},
+				Env: []corev1.EnvVar{
+					{
+						Name:  "ACTION_UID",
+						Value: string(backup.GetUID()),
+					},
+					{
+						Name:  "BACKUP_SNAPSHOT_ID",
+						Value: strconv.Itoa(*backup.Status.SnapID),
+					},
+					{
+						Name:  "BACKUP_TARGET_PVC_NAME",
+						Value: backup.Spec.PVC,
+					},
+					{
+						Name:  "BACKUP_TARGET_PVC_NAMESPACE",
+						Value: backup.Spec.PVCNamespace,
+					},
+					{
+						Name:  "BACKUP_TARGET_PVC_UID",
+						Value: labels[labelBackupTargetPVCUID],
+					},
+				},
+				Image:           r.podImage,
+				ImagePullPolicy: corev1.PullIfNotPresent,
+				VolumeMounts: []corev1.VolumeMount{
+					{
+						MountPath: nlv.VolumePath,
+						Name:      "fin-volume",
+					},
+				},
+			},
+		}
+
+		job.Spec.Template.Spec.Volumes = []corev1.Volume{
+			{
+				Name: "fin-volume",
+				VolumeSource: corev1.VolumeSource{
+					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+						ClaimName: finVolumePVCName(backup),
+					},
+				},
+			},
+		}
+
+		return nil
+	}); err != nil {
+		return err
+	}
 	return nil
 }
 
+//nolint:dupl
 func (r *FinBackupReconciler) createOrUpdateCleanupJob(ctx context.Context, backup *finv1.FinBackup) error {
-	// TODO: Implement the cleanup job creation logic.
+	var job batchv1.Job
+	job.SetName(cleanupJobName(backup))
+	job.SetNamespace(r.cephClusterNamespace)
+	if _, err := ctrl.CreateOrUpdate(ctx, r.Client, &job, func() error {
+		labels := job.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		labels["app.kubernetes.io/name"] = labelAppNameValue
+		labels["app.kubernetes.io/component"] = labelComponentCleanupJob
+		job.SetLabels(labels)
+
+		owners := []metav1.OwnerReference{
+			*metav1.NewControllerRef(backup, finv1.GroupVersion.WithKind("FinBackup")),
+		}
+		job.SetOwnerReferences(owners)
+
+		job.Spec.BackoffLimit = ptr.To(int32(maxJobBackoffLimit))
+
+		job.Spec.Template.Spec.NodeName = backup.Spec.Node
+
+		job.Spec.Template.Spec.SecurityContext = &corev1.PodSecurityContext{
+			FSGroup:      ptr.To(int64(10000)),
+			RunAsGroup:   ptr.To(int64(10000)),
+			RunAsNonRoot: ptr.To(true),
+			RunAsUser:    ptr.To(int64(10000)),
+		}
+
+		job.Spec.Template.Spec.ServiceAccountName = "fin-cleanup-job"
+
+		job.Spec.Template.Spec.RestartPolicy = corev1.RestartPolicyOnFailure
+
+		job.Spec.Template.Spec.Containers = []corev1.Container{
+			{
+				Name:    "cleanup",
+				Command: []string{"/manager"},
+				Args:    []string{"cleanup"},
+				Env: []corev1.EnvVar{
+					{
+						Name:  "ACTION_UID",
+						Value: string(backup.GetUID()),
+					},
+					{
+						Name:  "BACKUP_SNAPSHOT_ID",
```

Comment:
```suggestion
						Name:  "TARGET_SNAPSHOT_ID",
```

ref.
https://github.com/cybozu-go/fin/blob/107d1ff79842131db1fc72d18472fe47b25c12f0/cmd/cleanup.go#L63